src/internal/operators/bufferTime.ts

```diff
@@ -231,23 +239,26 @@ function dispatchBufferTimeSpanOnly(this: SchedulerAction<any>, state: any) {
     state.context = subscriber.openContext();
     state.context.closeAction = this.schedule(state, state.bufferTimeSpan);
   }
+
+  state.subscriber.destination.next(prevContext.buffer);
 }
 
 function dispatchBufferCreation<T>(this: SchedulerAction<DispatchCreateArg<T>>, state: DispatchCreateArg<T>) {
-  const { bufferCreationInterval, bufferTimeSpan, subscriber, scheduler } = state;
+  const { bufferCreationInterval, bufferTimeSpan, subscriber, scheduler, destination } = state;
   const context = subscriber.openContext();
   const action = <SchedulerAction<DispatchCreateArg<T>>>this;
   if (!subscriber.closed) {
     subscriber.add(context.closeAction = scheduler.schedule<DispatchCloseArg<T>>(
       dispatchBufferClose as any,
       bufferTimeSpan,
-      { subscriber, context }
+      { subscriber, context, destination }
     ));
     action.schedule(state, bufferCreationInterval!);
   }
 }
 
 function dispatchBufferClose<T>(arg: DispatchCloseArg<T>) {
-  const { subscriber, context } = arg;
+  const { subscriber, context, destination } = arg;
   subscriber.closeContext(context);
+  destination.next(context.buffer);
```

Comment:
I don't like that the logic is scattered around this big file, lots of indirection & my changes make it worse. One idea for future improvement to this operator would be to rewrite if from the ground up, reactively, by building on the `buffer(), delay(), count()` operators or something. Right now the implementation is very imperative which makes it not easy to reason about.